ios/RNSScreen.mm

```diff
@@ -964,6 +989,73 @@ - (void)viewDidLayoutSubviews
   }
 }
 
+- (void)dismissViewControllerAnimated:(BOOL)flag completion:(void (^)())completion
+{
+  [super dismissViewControllerAnimated:flag completion:completion];
+  BOOL isPresentedAsNativeModal = self.screenView.isPresentedAsNativeModal;
+
+  [self recalculateHeaderHeightIsModal:isPresentedAsNativeModal];
```

Comment:
I think we could move the `self.screenView.isPresentedAsNativeModal` here directly since it is quite descriptive by itself